ui/app/pages/settings/contact-list-tab/view-contact/view-contact.component.js

```diff
@@ -48,28 +66,41 @@ export default class ViewContact extends PureComponent {
           </div>
           <div className="address-book__view-contact__group">
             <div className="address-book__view-contact__group__label">
-              { t('ethereumPublicAddress') }
+              {t('ethereumPublicAddress')}
             </div>
             <div className="address-book__view-contact__group__value">
-              <div
-                className="address-book__view-contact__group__static-address"
-              >
-                { quadSplit(checkSummedAddress) }
+              <div className="address-book__view-contact__group__static-address">
+                {quadSplit(checkSummedAddress)}
               </div>
-              <button
-                className="address-book__view-contact__group__static-address--copy-icon"
-                onClick={() => copyToClipboard(checkSummedAddress)}
+              <Tooltip
+                wrapperClassName="address-book__tooltip-wrapper"
+                position="bottom"
+                title={
+                  this.state.copied
+                    ? t('copiedExclamation')
+                    : t('copyToClipboard')
+                }
               >
-                <Copy size={20} color="#3098DC" />
-              </button>
+                <button
+                  className="address-book__view-contact__group__static-address--copy-icon"
+                  onClick={() => {
+                    const { checkSummedAddress } = this.props
```

Comment:
Nit: It would be better to avoid shadowed variables if possible. In this case there's no need to declare another `checkSummedAddress` variable here, as there's already one in-scope.

```suggestion
```